src/System.IO/src/System/IO/StreamReader.cs

```diff
@@ -480,6 +521,14 @@ private void DetectEncoding()
                 CompressBuffer(3);
                 changedEncoding = true;
             }
+            else if (_byteLen >= 4 && _byteBuffer[0] == 0 && _byteBuffer[1] == 0 &&
+                _byteBuffer[2] == 0xFE && _byteBuffer[3] == 0xFF)
+            {
+                // Big Endian UTF32
+                _encoding = new UTF32Encoding(true, true);
```

Comment:
Nit: `new UTF32Encoding(bigEndian: true, byteOrderMark: true);`

Alternatively, the following would allow caching/reusing the `UTF32Encoding` instance, based on the current code in coreclr:

```c#
const int CodePageUTF32BE = 12001;
_encoding = Encoding.GetEncoding(CodePageUTF32BE);
```